packages/connect-thegraph/src/connector.ts

```diff
@@ -39,58 +36,60 @@ export default class ConnectorTheGraph extends GraphQLWrapper
   }
 
   async rolesForAddress(appAddress: string): Promise<Role[]> {
-    const result = await this.performQuery(queries.ROLE_BY_APP_ADDRESS, {
-      appAddress: appAddress.toLowerCase(),
-    })
-
-    const datas = this.parseQueryResult(parseRoles, result)
-
-    return datas.map((data: RoleData) => {
-      return new Role(data, this)
-    })
```

Comment:
I created a new low level function `performQueryWithParser(...)` which simplifies connector functions. Parsers now return the final objects, so all the parsing is delegated to the parsers, as it should.

Reply:
Great abstraction